From cd0668cb3b1eae72d933ba110ac8782a827a983d Mon Sep 17 00:00:00 2001
From: Gerard Andre <gerard.andre@intel.com>
Date: Tue, 14 Apr 2015 11:45:29 +0200
Subject: [PATCH] Check get_bat_prop_cache in trigger_algo

In trigger_algo, if get_bat_prop_cache does not
find the property, trigger_algo uses the value
found on its stack.

Change-Id: I125095c221cf14790fb454cd8d0e31dc0edccc53
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-29853
Signed-off-by: Jean-Christophe PINCE <jean-christophe.pince@intel.com>
Reviewed-on: https://android.intel.com:443/351787
---
 drivers/power/power_supply_charger.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/power/power_supply_charger.c b/drivers/power/power_supply_charger.c
index 73dd95c4e0e5..5fa75fe424c2 100644
--- a/drivers/power/power_supply_charger.c
+++ b/drivers/power/power_supply_charger.c
@@ -645,7 +645,7 @@ static int trigger_algo(struct power_supply *psy)
 	struct batt_props bat_prop;
 	struct charging_algo *algo;
 	struct ps_batt_chg_prof chrg_profile;
-	int cnt;
+	int cnt, ret;
 
 	if (psy->type != POWER_SUPPLY_TYPE_BATTERY)
 		return 0;
@@ -656,7 +656,12 @@ static int trigger_algo(struct power_supply *psy)
 	}
 
 
-	get_bat_prop_cache(psy, &bat_prop);
+	ret = get_bat_prop_cache(psy, &bat_prop);
+	if (ret) {
+		pr_err("%s:Error in getting the battery property of %s!!\n",
+							__func__, psy->name);
+		return ret;
+	}
 
 	algo = power_supply_get_charging_algo(psy, &chrg_profile);
 	if (!algo) {
-- 
1.9.1

